Realm/Realm/Realm.cs

```diff
@@ -1364,6 +1359,46 @@ internal void ExecuteOutsideTransaction(Action action)
 
         #endregion Transactions
 
+        internal class RealmMetadata
+        {
+            private readonly Dictionary<string, RealmObjectBase.Metadata> stringToRealmObjectMetadataDict;
+            private readonly Dictionary<TableKey, RealmObjectBase.Metadata> tableKeyToRealmObjectMetadataDict;
+
+            public IEnumerable<RealmObjectBase.Metadata> Values => stringToRealmObjectMetadataDict.Values;
+
+            public RealmMetadata(IEnumerable<RealmObjectBase.Metadata> objectsMetadata)
+            {
+                stringToRealmObjectMetadataDict = new Dictionary<string, RealmObjectBase.Metadata>();
+                tableKeyToRealmObjectMetadataDict = new Dictionary<TableKey, RealmObjectBase.Metadata>();
+
+                foreach (var objectMetadata in objectsMetadata)
+                {
+                    stringToRealmObjectMetadataDict[objectMetadata.Schema.Name] = objectMetadata;
+                    tableKeyToRealmObjectMetadataDict[objectMetadata.TableKey] = objectMetadata;
+                }
+            }
+
+            public bool TryGetValue(string objectType, out RealmObjectBase.Metadata metadata)
+            {
+                return stringToRealmObjectMetadataDict.TryGetValue(objectType, out metadata);
+            }
+
+            public bool TryGetValue(TableKey tablekey, out RealmObjectBase.Metadata metadata)
+            {
+                return tableKeyToRealmObjectMetadataDict.TryGetValue(tablekey, out metadata);
+            }
```

Comment:
These should be expression-bodied

Reply:
TODO: research the difference between expression bodied methods and regular ones.